lib/Doctrine/ODM/MongoDB/Persisters/DocumentPersister.php

```diff
@@ -1493,10 +1498,17 @@ private function getWriteOptions(array $options = []): array
         $defaultOptions  = $this->dm->getConfiguration()->getDefaultCommitOptions();
         $documentOptions = [];
         if ($this->class->hasWriteConcern()) {
-            $documentOptions['w'] = $this->class->getWriteConcern();
+            $documentOptions['writeConcern'] = new WriteConcern($this->class->getWriteConcern());
+        }
+
+        $writeOptions = array_merge($defaultOptions, $documentOptions, $options);
+        if (array_key_exists('w', $writeOptions)) {
+            @trigger_error(sprintf('The "w" option as commit option is deprecated, please pass "%s" object in "writeConcern" option.', WriteConcern::class), E_USER_DEPRECATED);
```

Comment:
My apologies, this comment was swallowed in my earlier review:

```suggestion
            trigger_deprecation(
                'doctrine/mongodb-odm',
                '2.2',
                'The "w" option as commit option is deprecated, please pass "%s" object in "writeConcern" option.',
                WriteConcern::class
            );
```